static/app/components/search/index.tsx

```diff
@@ -102,166 +104,147 @@ type Props = WithRouterProps<{orgId: string}> & {
+      defaultHighlightedIndex={0}
+      onSelect={handleSelectItem}
+      closeOnSelect={props.closeOnSelect ?? DefaultSearchProps.closeOnSelect}
+    >
+      {({getInputProps, getItemProps, isOpen, inputValue, highlightedIndex}) => {
+        const searchQuery = inputValue.toLowerCase().trim();
+        const isValidSearch = inputValue.length >= props.minSearch;
+
+        debouncedSaveQueryMetrics(searchQuery);
+
+        const renderItem =
+          typeof props.renderItem === 'function'
+            ? props.renderItem
+            : DefaultSearchProps.renderItem;
+
+        return (
+          <SearchWrapper>
+            {props.renderInput({getInputProps})}
+
+            {isValidSearch && isOpen ? (
+              <SearchSources
+                searchOptions={props.searchOptions}
+                query={searchQuery}
+                params={props.params}
+                sources={props.sources ?? DefaultSearchProps.sources}
+              >
+                {({isLoading, results, hasAnyResults}) => (
+                  <DropdownBox className={props.dropdownStyle}>
+                    {isLoading ? (
+                      <LoadingWrapper>
+                        <LoadingIndicator mini hideMessage relative />
+                      </LoadingWrapper>
+                    ) : !hasAnyResults ? (
+                      <EmptyItem>{t('No results found')}</EmptyItem>
+                    ) : (
+                      results.slice(0, props.maxResults).map((resultObj, index) => {
+                        return React.cloneElement(
+                          renderItem({
                             index,
-                            highlightedIndex,
-                            getItemProps,
-                          })
-                        )}
-                      {!isLoading && !hasAnyResults && (
-                        <EmptyItem>{t('No results found')}</EmptyItem>
-                      )}
-                      {!isLoading && resultFooter && (
-                        <ResultFooter>{resultFooter}</ResultFooter>
-                      )}
-                    </DropdownBox>
-                  )}
-                </SearchSources>
-              ) : null}
-            </SearchWrapper>
-          );
-        }}
-      </AutoComplete>
-    );
-  }
+                            item: resultObj.item,
+                            matches: resultObj.matches,
+                            highlighted: index === highlightedIndex,
+                            itemProps: getItemProps({
+                              item: resultObj.item,
+                              index,
+                            }),
+                          }),
+                          {key: `${resultObj.item.title}-${index}`}
+                        );
+                      })
+                    )}
+                    {!isLoading && props.resultFooter ? (
+                      <ResultFooter>{props.resultFooter}</ResultFooter>
+                    ) : null}
```

Comment:
```suggestion
                    {!isLoading && props.resultFooter && (
                      <ResultFooter>{props.resultFooter}</ResultFooter>
                    )}
```

Reply:
@priscilawebdev for render fn's, I usually always default to a ternary expression to avoid falsy values from being rendered (https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)

> Note that returning a falsy expression will still cause the element after && to be skipped but will return the falsy expression. In the example below, <div>0</div> will be returned by the render method.

I think this is more future proof way of rendering, that prevents us from leaking into the dom in case we decide to refactor/change props